File: frontend/src/app/features/project/models/project.dto.ts
import {UserDto} from "../../user/models/user.dto";

export interface ProjectDto {
    projectId: number;
    projectName: string;
    projectDescription: string;
    projectDateFrom: Date;
    projectDateTo: Date;
    projectObjective: string;
    projectCloseMessage: string;
    projectEndDate: Date | null;
    projectOwners: UserDto[];
    projectModerators: UserDto[];
    projectMembers: UserDto[];
    finishedTasks: number;
    totalTasks: number;
}

File: backend/src/main/kotlin/bo/edu/umsa/backend/dto/ProjectDto.kt
package bo.edu.umsa.backend.dto

import java.sql.Timestamp

data class ProjectDto(
    val projectId: Int,
    val projectName: String,
    val projectDescription: String,
    val projectObjective: String,
    val projectCloseMessage: String,
    val projectDateFrom: Timestamp,
    val projectDateTo: Timestamp,
    val projectEndDate: Timestamp?,
    val projectOwners: List<UserPartialDto>,
    val projectModerators: List<UserPartialDto>,
    val projectMembers: List<UserPartialDto>,
    val finishedTasks: Int,
    val totalTasks: Int,
)
